model_analyzer/config/input/config_command_profile.py

```diff
@@ -594,6 +595,10 @@ def _autofill_values(self):
         config values.
         """
 
+        min_metric_tags = DEFAULT_METRICS
+        extra_metric_tags = list(set(self.metrics) - set(min_metric_tags))
+        self._fields["metrics"].set_value(min_metric_tags + extra_metric_tags)
```

Comment:
This is good for now. Ideally we want flexibility on what the user is capturing, but the model analyzer requires there be some metrics like throughput. This makes things tricky because in analysis we have to make sure that the user does not try to prioritize the results by a metric that was not captured. 

One solution I can think of for this is to store the captured metric list to the checkpoint as profile metadata. This must somehow then be used to check the objectives and constraints sections of the ConfigCommandAnalyze.

To make this easier for 21.08, maybe we don't give the user full control over the metrics to be collected, but instead going back to what you did, add an argument `--collect-cpu-metrics`.  The logic you have in this PR can remain the same. This flag will be false by default, but if true or there is a model that is cpu_only, `cpu_used_ram` and `cpu_available_ram` will be added to the metric tags internally.

Reply:
I agree this is good for now, for 21.08.

The user actually does not have the ability to control flags other than `cpu_used_ram` and `cpu_available_ram`. It is because the default metrics - the same as previously collected metrics except `cpu_used_ram` and `cpu_available_ram` - will always be collected regardless of what the user specifies. If the user adds `cpu_used_ram` and/or `cpu_available_ram`, then CPU monitor will run and collect the requested CPU metrics.

I think this is good placeholder for future changes. Currently, if a model is `cpu_only` or no GPU is visible, then a `logging.info` will be printed to remind the user when none of the CPU metric flag is set. (Instead of automatically adding the flag.)

For the `ConfigCommandAnalyze`, instead of capturing the metric list, the metric list is determined by checking the collected measurements. This removes a possible corner case where the user runs `profile` twice with a different `--metrics` flag.